components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/WriterGroupMessageSource.cs

```diff
@@ -275,7 +301,30 @@ private sealed class DataSetWriterSubscription : IDisposable {
                         notification = snapshot;
                     }
                 }
-                CallMessageReceiverDelegates(sender, sequenceNumber, notification, true);
+                CallMessageReceiverDelegates(sender, sequenceNumber, notification);
+            }
+
+            /// <summary>
+            /// Handle subscription event diagnostics change messages
+            /// </summary>
+            /// <param name="sender"></param>
+            /// <param name="notificationCount"></param>
+            private void OnSubscriptionEventDiagnosticsChangedAsync(object sender, int notificationCount) {
```

Comment:
Why is this function called async?

Reply:
Original functions was, but they also did asynchronous call. Thus doesn't so fixed.